src/index.ts

```diff
@@ -1,24 +1,51 @@
 import { ensureString } from "ensure-string";
 import mlArrayMax from "ml-array-max";
-import uniqueXFunction from "ml-arrayxy-uniquex";
-import { xIsMonotone } from "ml-spectra-processing";
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const uniqueXFunction = require("ml-arrayxy-uniquex");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { xIsMonotone } = require("ml-spectra-processing");
```

Comment:
ditto

Reply:
require is used to avoid typescript erros (the packages are  javascript)

Reply:
I checked out your repo and it compiles without errors with an `import`. What errors are you having? I would have expected there to be "no implicit any" but there actually isn't, I'm not sure why because the ts `strict` mode is on

When building the esm lib typescript keeps the `require` mixed with `import` and I don't think that can be executed by nodejs when it runs in esm mode. So ASAIK it could be problematic to keep those `require` statements.

@targos what is your recommandation, you know much more than me about those things ^^

Reply:
If there is an error because the package does not provide TS definitions, the solution is to add a `// @ts-expect-error` comment. Please do not use `require()` in TypeScript.

Reply:
error fixed thank you

Reply:
it was because the any type while importing js package